client/my-sites/site-settings/composing/publish-confirmation.jsx

```diff
@@ -1,82 +1,21 @@
-import { localize } from 'i18n-calypso';
-import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
-import { bindActionCreators } from 'redux';
+import { useTranslate } from 'i18n-calypso';
 import FormFieldset from 'calypso/components/forms/form-fieldset';
 import FormLabel from 'calypso/components/forms/form-label';
 import FormSettingExplanation from 'calypso/components/forms/form-setting-explanation';
-import { saveConfirmationSidebarPreference } from 'calypso/state/editor/actions';
-import { isConfirmationSidebarEnabled } from 'calypso/state/editor/selectors';
-import { isFetchingPreferences } from 'calypso/state/preferences/selectors';
-import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 
-class PublishConfirmation extends Component {
-	constructor( props ) {
-		super( props );
-		this.state = { isToggleOn: props.publishConfirmationEnabled };
-		this.handleToggle = this.handleToggle.bind( this );
-	}
-
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( nextProps.publishConfirmationEnabled !== this.state.isToggleOn ) {
-			this.setState( { isToggleOn: nextProps.publishConfirmationEnabled } );
-		}
-	}
-
-	handleToggle() {
-		const { siteId, savePublishConfirmationPreference } = this.props;
-		const isToggleOn = ! this.state.isToggleOn;
-		this.setState( { isToggleOn: isToggleOn } );
-		savePublishConfirmationPreference( siteId, isToggleOn );
-	}
-
-	render() {
-		const { translate } = this.props;
-
-		return (
-			<FormFieldset>
-				<FormLabel>{ translate( 'Show publish confirmation' ) }</FormLabel>
-				<FormSettingExplanation>
-					{ translate(
-						'The Block Editor handles the Publish confirmation setting. ' +
-							'To enable it, go to Options under the Ellipses menu in the Editor ' +
-							'and check "Enable Pre-publish checks."'
-					) }
-				</FormSettingExplanation>
-			</FormFieldset>
-		);
-	}
+export default function PublishConfirmation() {
+	const translate = useTranslate();
+
+	return (
+		<FormFieldset>
+			<FormLabel>{ translate( 'Show publish confirmation' ) }</FormLabel>
+			<FormSettingExplanation>
+				{ translate(
+					'The Block Editor handles the Publish confirmation setting. ' +
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'The Block Editor handles the Publish confirmation setting. To enable it, go to Options under the Ellipses menu in the Editor and check "Enable Pre-publish checks."' )` *ES Score: 10*

Reply:
Good suggestion @a8ci18n, but we're intentionally fixing that typo.